app/src/main/java/ch/epfl/culturequest/ui/scan/ScanFragment.java

```diff
@@ -156,12 +150,29 @@ public void onSurfaceTextureUpdated(@NonNull SurfaceTexture surfaceTexture) {
     };
 
     private final View.OnClickListener cancelButtonListener = view -> {
-        loadingAnimation.stopLoading();
-        scanningLayout.setVisibility(View.GONE);
-
-        // Cancel the current processing if it exists
-        if(currentProcessing != null)
-            currentProcessing.cancel(true);
```

Comment:
I didnt remove it, i put it in the else of the check if user has connection. I added the if so that if the user presses on cancel when they're offline, we warn them that they will lose their scan if they cancel. But users can still cancel like before if they're online and click on cancel

Reply:
Oh okay I see. Good then!